pkg/log/log.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package log contains logging related functionality
+package log
+
+import (
+	"io"
+)
+
+// Logger is a interface that should be used for all logging inside kind/pkg.
+// Consumers of kind/pkg such as kind/cmd should define an object that satisfies
+// this interface.
```

Comment:
Let's also add a TODO to consider klog / `logr` when it has more traction. (EG logr support in klog... ) @munnerz was working on this, but I don't think it will be ready for a while.